p2p/module.go

```diff
@@ -199,3 +217,99 @@ func (m *p2pModule) handleNetworkMessage(networkMsgData []byte) {
 
 	m.GetBus().PublishEventToBus(&event)
 }
+
+func configureBootstrapNodes(p2pCfg *configs.P2PConfig, m *p2pModule) error {
+	var csvReader *csv.Reader
+	customBootstrapNodesCsv := strings.Trim(p2pCfg.BootstrapNodesCsv, " ")
+	if customBootstrapNodesCsv == "" {
+		csvReader = csv.NewReader(strings.NewReader(defaults.DefaultP2PBootstrapNodesCsv))
+	} else {
+		csvReader = csv.NewReader(strings.NewReader(customBootstrapNodesCsv))
+	}
+	bootStrapNodes, err := csvReader.Read()
+	if err != nil {
+		return fmt.Errorf("error parsing bootstrap nodes: %w", err)
+	}
+	for _, node := range bootStrapNodes {
+		if !isValidHostnamePort(node) {
+			return fmt.Errorf("invalid bootstrap node: %s", node)
+		}
+	}
+	m.bootstrapNodes = bootStrapNodes
+	return nil
+}
+
+func isValidHostnamePort(str string) bool {
+	pattern := regexp.MustCompile(`^(https?)://([a-zA-Z0-9.-]+):(\d{1,5})$`)
+	matches := pattern.FindStringSubmatch(str)
+	if len(matches) != 4 {
+		return false
+	}
+	protocol := matches[1]
```

Comment:
It's not zero indexisble?

Reply:
the 0th index is the (outermost) group that matches the whole regex

Reply:
More context: 

https://pkg.go.dev/regexp#Regexp.FindStringSubmatch

> If 'Submatch' is present, the return value is a slice identifying the successive submatches of the expression. Submatches are matches of parenthesized subexpressions (also known as capturing groups) within the regular expression, numbered from left to right in order of opening parenthesis. Submatch 0 is the match of the entire expression, submatch 1 is the match of the first parenthesized subexpression, and so on.